host/include/uhd/types/endianness.hpp

```diff
@@ -20,6 +20,21 @@
 
 #include <uhd/config.hpp>
 
+/******************************************************************************
+ * Detect host endianness
+ *****************************************************************************/
+#include <boost/predef/other/endian.h>
+
+// In Boost 1.55, the meaning of the macros changed. They are now always
+// defined, but don't always have the same value.
+#if BOOST_ENDIAN_BIG_BYTE
```

Comment:
With this change, Boost 1.54 and older is no longer supported. Probably best to just remove 1.46...1.54 from Boost_ADDITIONAL_VERSIONS.

Reply:
We have a different change (backported stuff from master). I'll chat with the 3.9 maintainer to get it merged.